pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -189,6 +190,48 @@ func EnsureProjectReferenceStatusCreating(adapter *ReferenceAdapter) (gcputil.Op
 	return gcputil.StopProcessing()
 }
 
+func serviceNameAlreadyGenerated(projectReference *gcpv1alpha1.ProjectReference) bool {
+	osdServiceAccountNameDefaultPrefix := serviceAccountNameTemplate("")
+	serviceAccountName := projectReference.Spec.ServiceAccountName
+	return strings.HasPrefix(serviceAccountName, osdServiceAccountNameDefaultPrefix) &&
+		len(serviceAccountName) > len(osdServiceAccountNameDefaultPrefix)
+}
+
+func serviceNameAlreadyFilled(projectReference *gcpv1alpha1.ProjectReference) bool {
+	return projectReference.Spec.ServiceAccountName == osdServiceAccountNameDefault
+}
+func serviceAccountNameTemplate(suffix string) string {
+	return fmt.Sprintf("%s-%s", osdServiceAccountNameDefault, suffix)
+}
+
+func EnsureServiceAccountNameMigration(adapter *ReferenceAdapter) (gcputil.OperationResult, error) {
+	adapter.logger.V(1).Info("enter EnsureServiceAccountNameMigration")
+	if adapter.ProjectReference.Status.State == gcpv1alpha1.ProjectReferenceStatusReady &&
+		adapter.ProjectReference.Spec.ServiceAccountName == "" {
+
+		adapter.ProjectReference.Spec.ServiceAccountName = osdServiceAccountNameDefault
+		err := adapter.kubeClient.Update(context.TODO(), adapter.ProjectReference)
+		return gcputil.RequeueOnErrorOrStop(err)
+	}
+	return gcputil.ContinueProcessing()
+}
```

Comment:
Plan is to remove this once all environments are migrated?

Reply:
yes